atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/RowColumnRangeExtractor.java

```diff
@@ -69,10 +70,14 @@ public RowColumnRangeResult(Map<byte[], LinkedHashMap<Cell, Value>> results,
     private final Map<byte[], Integer> rowsToRawColumnCount = Maps.newHashMap();
     private final Set<byte[]> emptyRows = Sets.newHashSet();
 
-    public void extractResults(Map<ByteBuffer, List<ColumnOrSuperColumn>> colsByKey,
-                                                long startTs) {
+    public void extractResults(Iterable<byte[]> canonicalRows,
+                               Map<ByteBuffer, List<ColumnOrSuperColumn>> colsByKey,
+                               long startTs) {
+        // Make sure returned maps are keyed by the given rows
+        Map<Sha256Hash, byte[]> canonicalRowsByHash = Maps.uniqueIndex(canonicalRows, Sha256Hash::computeHash);
```

Comment:
Should probably document that if the canonicalRows is not distinct Set<byte[]> that this will throw